java/ql/test/experimental/query-tests/security/CWE-918/URLClassLoaderSSRF.java

```diff
@@ -0,0 +1,71 @@
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+
+public class URLClassLoaderSSRF extends HttpServlet {
+
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        try {
+            String url = request.getParameter("uri");
+            URI uri = new URI(url);
+            URLClassLoader urlClassLoader = new URLClassLoader(new URL[]{uri.toURL()});
+            Class<?> test = urlClassLoader.loadClass("test");
+        } catch (Exception e) {
+            // Ignore
+        }
+    }
+
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        try {
+            String url = request.getParameter("uri");
+            URI uri = new URI(url);
+            URLClassLoader urlClassLoader = URLClassLoader.newInstance(new URL[]{uri.toURL()});
+            Class<?> test = urlClassLoader.loadClass("test");
+        } catch (Exception e) {
+            // Ignore
+        }
+    }
+
+    protected void doPut(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        try {
+            String url = request.getParameter("uri");
+            URI uri = new URI(url);
+            URLClassLoader urlClassLoader = new URLClassLoader(new URL[]{uri.toURL()});;
+            urlClassLoader.findResource("test");
+        } catch (Exception e) {
+            // Ignore
+        }
+    }
+
+    protected void doDelete(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        try {
+            String url = request.getParameter("uri");
+            URI uri = new URI(url);
+            URLClassLoader urlClassLoader = URLClassLoader.newInstance(new URL[]{uri.toURL()});
+            urlClassLoader.getResourceAsStream("test");
+        } catch (Exception e) {
+            // Ignore
+        }
+    }
+
+    protected void doPatch(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        try {
+            String url = request.getParameter("uri");
+            URI uri = new URI(url);
+            URLClassLoader urlClassLoader = new URLClassLoader(new URL[]{new URL("aaaaa")});
```

Comment:
Is it deliberate that you've used "aaaaa" here to avoid getting a result? Perhaps add a comment to make this completely clear.